arm/authorization/client.go

```diff
@@ -0,0 +1,759 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator 0.11.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package authorization
+
+import (
+	"fmt"
+	"github.com/azure/go-autorest/autorest"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	ApiVersion             = "2015-01-01"
+	DefaultBaseUri         = "https://management.azure.com"
+	DefaultPollingDuration = 10 * time.Minute
+)
+
+type AuthorizationClient struct {
+	autorest.Client
+	BaseUri        string
+	SubscriptionId string
+}
+
+func New(subscriptionId string) *AuthorizationClient {
+	return NewWithBaseUri(DefaultBaseUri, subscriptionId)
+}
+
+func NewWithBaseUri(baseUri string, subscriptionId string) *AuthorizationClient {
+	client := &AuthorizationClient{BaseUri: baseUri, SubscriptionId: subscriptionId}
+	client.PollingMode = autorest.PollUntilDuration
+	client.PollingDuration = DefaultPollingDuration
+	return client
+}
+
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
@brendandixon if they're indeed different clients and you're having trouble skimming long files why not split different client types into different source files such as:

- compute/availabilitysets.go
- compute/availabilitysets.go
- compute/virtualmachineimages.go
- compute/virtualmachineimages_entities.go